azure-proto-core/ArmClient.cs

```diff
@@ -167,143 +115,96 @@ await foreach (var location in ListAvailableLocationsAsync(subscriptionId, resou
+        public T GetResourceOperationsBase<T>(TrackedResource resource) where T : TrackedResource
         {
             return Activator.CreateInstance(typeof(T), ClientContext, resource) as T;
         }
 
-        public T GetResourceOperationsBase<T>(ResourceIdentifier resource)
-            where T : OperationsBase
+        public T GetResourceOperationsBase<T>(ResourceIdentifier resource) where T : OperationsBase
         {
             return Activator.CreateInstance(typeof(T), ClientContext, resource) as T;
         }
 
-        public T GetResourceOperationsBase<T>(string subscription, string resourceGroup, string name)
-            where T : OperationsBase
+        public T GetResourceOperationsBase<T>(string subscription, string resourceGroup, string name) where T : OperationsBase
         {
             return null;
         }
 
-        public ArmResponse<TOperations> CreateResource<TContainer, TOperations, TResource>(
-            string subscription,
-            string resourceGroup,
-            string name,
-            TResource model,
-            Location location = default)
+        public ArmResponse<TOperations> CreateResource<TContainer, TOperations, TResource>(string subscription, string resourceGroup, string name, TResource model, azure_proto_core.Location location = default)
             where TResource : TrackedResource
             where TOperations : ResourceOperationsBase<TOperations, TResource>
             where TContainer : ResourceContainerOperations<TOperations, TResource>
         {
             if (location == null)
             {
-                location = Location.Default;
+                location = azure_proto_core.Location.Default;
             }
 
-            var container = Activator.CreateInstance(
-                    typeof(TContainer),
-                    ClientContext,
-                    new ArmResource($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location)) as
-                TContainer;
+            TContainer container = Activator.CreateInstance(typeof(TContainer), ClientContext, new ArmResource($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location)) as TContainer;
 
             return container.Create(name, model);
         }
 
         /// <summary>
-        ///     Fill in the default subscription in the simple case (passed in, or only one subscription available)
+        /// Fill in the default subscription in the simple case (passed in, or only one subscription available)
         /// </summary>
         /// <param name="token"></param>
         /// <returns></returns>
-        internal async Task<string> GetDefaultSubscription(CancellationToken token = default)
+        internal async Task<string> GetDefaultSubscription(CancellationToken token = default(CancellationToken))
         {
-            var sub = DefaultSubscription?.Id?.Subscription;
+            string sub = DefaultSubscription?.Id?.Subscription;
             if (null == sub)
```

Comment:
>null [](start = 16, length = 4)

string.IsNullOrWhitespace()?

Reply:
Would not white space in this case be a case for an exception? If that happened, something went wrong.